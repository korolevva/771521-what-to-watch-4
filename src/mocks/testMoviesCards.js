export const moviesCards = [
  {
    id: 1,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    imagePreview: `img/bohemian-rhapsody.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    date: 2018,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury`,
    rating: 8.0,
    ratingCount: 305,
    director: `Bryan Singer`,
    stars: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    duration: 167,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    background: `https://placeimg.com/1300/552/tech`,
    imagePreview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    date: 2018,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    ratingCount: 423,
    director: `David Yates`,
    stars: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    duration: 90,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
