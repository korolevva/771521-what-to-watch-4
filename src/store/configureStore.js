import {createStore, compose, applyMiddleware} from "redux";
import {rootReducer} from "../reducers/index.js";
import {createAPI} from "../api.js";
import thunk from "redux-thunk";
import {Operation} from "../actions/dataActions.js";

const api = createAPI(() => { });

// export const store = createStore(
//     rootReducer,
//     window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
// );

export const store = createStore(
    rootReducer,
    compose(
        applyMiddleware(thunk.withExtraArgument(api)),
        window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
    )
);

store.dispatch(Operation.loadMovie());
store.dispatch(Operation.loadMovies());

