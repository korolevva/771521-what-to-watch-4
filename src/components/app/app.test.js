import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";
import {moviesCards} from "../../mocks/testMoviesCards.js";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducers/name-space.js";

const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.MOVIE_CARD]: {
    displayedMoviesCards: 8,
    selectedMovieCard: moviesCards[0],
    playingMovieCard: moviesCards[0],
  },
  [NameSpace.DATA]: {
    promoMovie: moviesCards[0],
    moviesCards,
  },
});


it(`Render App`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <App />,
    </Provider>,
    {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
