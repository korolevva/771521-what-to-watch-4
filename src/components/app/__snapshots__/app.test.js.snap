// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <div
    className="player"
  >
    <video
      autoPlay={false}
      className="player__video"
      controls={false}
      height="175"
      muted={false}
      poster="https://placeimg.com/270/410/arch/grayscale"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <button
      className="player__exit"
      onClick={[Function]}
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value="NaN"
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "NaN%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          00:00:00
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="21"
            viewBox="0 0 14 21"
            width="14"
          >
            <use
              xlinkHref="#pause"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          className="player__name"
        >
          Transpotting
        </div>
        <button
          className="player__full-screen"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>,
  ",",
]
`;
